docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,326 @@
+\htmlhr
+\chapterAndLabel{Resource Leak Checker for must-call obligations}{resource-leak-checker}
+
+The Resource Leak Checker guarantees that the program fulfills every object's
+must-call obligations before the object is de-allocated.
+
+A resource leak occurs when a program does not explicitly dispose of some finite
+underlying resource, such as a socket, file descriptor, or database connection.  To dispose
+of the resource, the program should call some method on an object.
+(De-allocating or garbage-collecting the object is not sufficient.)  For
+example, the program must call \<close()> on every object that implements the
+interface \<java.io.Closeable>.
+
+The Resource Leak Checker can check any property of the form ``the programmer
+must call methods \emph{M} on object \emph{O} before \emph{O} is
```

Comment:
An additional constraint is that Calling `M` once must be equivalent to calling `M` multiple times.   I don't know if that needs to be mentioned here or not.